### PR TITLE
feat: initial commit of Snapshot middleware

### PR DESCRIPTION
**Pull request by danleyb2**
### Summary by CodeRabbit

- New Feature: Introduced Snapshot middleware to format and forward requests from supported camera sources.
- New Feature: Implemented custom error handling with retry logic for HTTP requests.
- New Feature: Added Rollbar integration for error logging and monitoring.
- Documentation: Provided deployment and local development instructions in README.
- Test: Initial test suite for Cloudflare worker, ensuring basic functionality.
- Chore: Added `.editorconfig` and `.gitignore` for consistent coding style and repository cleanliness.

### CODE REVIEW

File: snapshot-middleware/src/rollbar.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+		let init = {
+			method: "POST",
+			body: JSON.stringify({ data: rollbarData }),
+			headers: {
+				"Content-type": "application/json",
+				"X-Rollbar-Access-Token": this.token,
+			},
+		};
+		return fetchWithRetry(rollbarUrl, init).then((response) => response.json());
```
[the remark this answers is not in the document]
Comment (danleyb2) [reply snapshot-middleware/src/rollbar.js:85]: this by design so it's reported by rollbar

File: snapshot-middleware/src/rollbar.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+const rollbarUrl = `https://api.rollbar.com/api/1/item/`;
+
+function Frame({ fileName, lineNumber, columnNumber, functionName, args }) {
+	const data = {};
+	data.filename = fileName;
+	data.lineno = lineNumber;
+	data.colno = columnNumber;
+	data.method = functionName;
+	data.args = args;
+	return data;
```
[the remark this answers is not in the document]
Comment (Pefington) [reply snapshot-middleware/src/rollbar.js:13]: nit

File: snapshot-middleware/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+npm run logs
+```
+
+## Local development
+- Create a `.dev.vars` with the env variables required
+- Run `npm run dev` in your terminal to start a development server
+- Open a browser tab at http://localhost:8787/ to see your worker in action
+- Test with curl
+    ```shell
+  	curl -vX POST http://localhost:8787/ -d @Survision.txt --header "Content-Type: application/json" --header "survision-serial-number: sv1-searial-1"
```
[the remark this answers is not in the document]
Comment (danleyb2) [reply snapshot-middleware/README.md:60]: You need to run the app `npm run dev` , Survision is just a text file with json data to be uploaded
[Survision.txt](https://github.com/user-attachments/files/17466189/Survision.txt)

[the remark this answers is not in the document]
Comment (danleyb2) [reply snapshot-middleware/README.md:65]: @Pefington `searial! is not a typo. It's just a test value

File: snapshot-middleware/test/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import {
+	env,
+	createExecutionContext,
+	waitOnExecutionContext,
+	SELF,
+} from "cloudflare:test";
+import { describe, it, expect } from "vitest";
+import worker from "../src";
+
+describe("Hello World worker", () => {
```
[the remark this answers is not in the document]
Comment (danleyb2) [reply snapshot-middleware/test/index.spec.js:10]: @Pefington This is already working for the customer and we have error logging to Rollbar. Let's add tests in a new PR.  FYI @marcbelmont

**Review by Pefington — APPROVED** [review]

**Review by marcbelmont — APPROVED** [review]
LGTM

nit: could you use 2 or 4 spaces for indentation?

**Comment by Pefington** [comment]
> LGTM
> 
> nit: could you use 2 or 4 spaces for indentation?

@danleyb2 Don't hesitate to install and use Prettier and eslint for JS/TS.

**Comment by marcbelmont** [comment]
@danleyb2 
don't forget to update the indentation.

**Review by marcbelmont — APPROVED** [review]

**Pull request merged**